src/lib/tls/credentials_manager.h

```diff
@@ -161,6 +161,17 @@ class BOTAN_PUBLIC_API(2, 0) Credentials_Manager {
       * key exists
       */
       virtual SymmetricKey psk(const std::string& type, const std::string& context, const std::string& identity);
+
+      /**
+       TODO: describe me
+       */
+      virtual std::vector<TLS::PresharedKey> tls13_preshared_keys_for(std::string_view host);
+
+      /**
+       TODO: describe me
+       */
```

Comment:
Make sure to mention that implementations need to cope with the fact that all identities provided to them might be resumption PSKs. I.e. the have to ignore _all_ members of the list and return `std::nullopt`. Same is true for `choose_session_to_resume()` (or similar name) in `Session_Manager`.